addons/account_facturx/models/ir_actions_report.py

```diff
@@ -16,17 +15,28 @@ def _post_pdf(self, save_in_attachment, pdf_content=None, res_ids=None):
             invoice = self.env['account.move'].browse(res_ids)
             if invoice.is_sale_document() and invoice.state != 'draft':
                 xml_content = invoice._export_as_facturx_xml()
-
-                # Add attachment.
-                reader_buffer = io.BytesIO(pdf_content)
+                metadata_content = self.env.ref('account_facturx.account_invoice_pdfa_3_facturx_metadata').render({
+                    'title': invoice.name,
+                    'date': fields.Date.context_today(self),
+                })
+                header = """<?xpacket begin="﻿" id="W5M0MpCehiHzreSzNTczkc9d"?>""".encode()
+                footer = """<?xpacket end="w"?>""".encode()
+                metadata = b"%s%s%s" % (header, metadata_content, footer)
+
+                reader_buffer = BytesIO(pdf_content)
                 reader = PdfFileReader(reader_buffer)
-                writer = PdfFileWriter()
+                writer = OdooPdfFileWriter()
                 writer.cloneReaderDocumentRoot(reader)
-                writer.addAttachment('factur-x.xml', xml_content)
-                buffer = io.BytesIO()
+
+                if literal_eval(self.env['ir.config_parameter'].sudo().get_param('edi.use_pdfa', 'False')):
```

Comment:
It is customary to use `tools.str2bool` rather than `literal_eval` for parsing boolean ICP values. That's nicer for users because they can use any of 0/1/yes/no/true/false/on/off case-insensitively with the expected result, rather than having to write Python literals exactly.

(it's also better because you don't want to evaluate arbitrary values, just yes/no, so you don't need the full power of literal_eval)